assets/src/edit-story/components/panels/pageAttachment/index.js

```diff
@@ -0,0 +1,167 @@
+  const [displayWarning, setDisplayWarning] = useState(false);
+
+  const { getLinksInAttachmentArea } = useElementsWithLinks();
+  const linksInAttachmentArea = getLinksInAttachmentArea();
+  const hasLinksInAttachmentArea = linksInAttachmentArea.length > 0;
+
+  // Stop displaying guidelines when unmounting.
+  useEffect(() => {
+    return () => {
+      setDisplayLinkGuidelines(false);
+    };
+  }, [hasLinksInAttachmentArea, setDisplayLinkGuidelines, url]);
+
+  // If we focus on the field and there are links in the area.
+  const onFocus = () => {
+    if (hasLinksInAttachmentArea && !url?.length) {
+      setDisplayWarning(true);
+      setDisplayLinkGuidelines(true);
+    }
+  };
+
+  // If the Page Attachment is added, stop displaying the warning.
+  useEffect(() => {
+    if (displayWarning && url?.length) {
+      setDisplayWarning(false);
+    }
+  }, [url, displayWarning]);
+
+  const updatePageAttachment = useCallback(
+    (value) => {
+      if (value.url) {
+        const urlWithProtocol = withProtocol(value.url);
+        const valid = isValidUrl(urlWithProtocol);
+        setIsInvalidUrl(!valid);
+      }
+      const _pageAttachment = {
+        ...pageAttachment,
+        ...value,
+      };
+      updateCurrentPageProperties({
+        properties: { pageAttachment: _pageAttachment },
+      });
+    },
+    [updateCurrentPageProperties, pageAttachment]
+  );
+
+  const [debouncedCtaUpdate] = useDebouncedCallback((value) => {
+    updatePageAttachment({ ctaText: value });
+  }, 300);
+
+  const [isInvalidUrl, setIsInvalidUrl] = useState(
+    !isValidUrl(withProtocol(url || ''))
+  );
+
+  const isDefault = _ctaText === defaultCTA;
+  return (
+    <SimplePanel
+      name="pageAttachment"
+      title={__('Page Attachment', 'web-stories')}
+    >
+      <LinkInput
+        description={__(
+          'Type an address to add a page attachment',
+          'web-stories'
+        )}
+        onChange={(value) => updatePageAttachment({ url: value })}
+        onFocus={onFocus}
+        value={url || ''}
+        clear
+        aria-label={__('Edit: Page Attachment link', 'web-stories')}
+      />
+
+      {displayWarning && (
+        <Row>
+          <Error>
+            {__(
+              'Links cannot reside below the dashed line when a page attachment is present. If you add a page attachment, your viewers will not be able to click on the link.',
+              'web-stories'
+            )}
+          </Error>
+        </Row>
+      )}
+
+      {Boolean(url) && !isInvalidUrl && (
+        <Row>
+          <ExpandedTextInput
+            onChange={(value) => {
+              // This allows smooth input value change without any lag.
+              _setCtaText(value);
+              debouncedCtaUpdate(value);
+            }}
+            onBlur={(atts = {}) => {
+              const { onClear } = atts;
+              if (onClear) {
+                updatePageAttachment({ ctaText: defaultCTA });
+              }
```

Comment:
Not sending current value upstream here will cause it to only update once debounce expires (which will be no more than .3s of course).

What we could do here is [cancel the latest debounce](https://github.com/xnimorz/use-debounce#cancel-maxwait-and-memoization) and directly invoke it to make sure that it syncs instantly when blurred, but that might also be overkill.

It would simplify the test though, as you could blur the input field and trust that the callback will have been invoked.

Playing a bit around with it, I actually think we should do this. Shouldn't be too tricky implement.

Reply:
I was 50-50 myself as well with the `onBlur` not being instant, changed now.